### PR TITLE
Handle hex for new cosmos-evm chains

### PR DESCRIPTION
**Pull request by mhagel**
## Link to Issue
Closes: #6584 
Closes: #6961 
Closes: #7325

## Description of Changes
- New EVM-compatible Cosmos chains will be recognized so keplr-ethereum or Metamask is used
- Refactored the Cosmos Proxy code for clarity and to eliminate unhandled promises.

## "How We Fixed It"

- Adds slip44 column to ChainNodes
  - 60 = evm-compatible node, 118 = standard cosmos coin type
  - queries cosmos.directory for slip44 if not in our DB (handle new chains automatically)
  - Use this differentiator to choose Keplr-Ethereum vs Keplr wallet api
- Reduces sensitivity of Node Liveness system - only flags failed nodes if querying, not making transactions

## Test Plan

- CA (click around) tested on local and:
- run migration
For 6961:
  - at /evmos-dev, [sign in and create a proposal](https://github.com/hicommonwealth/commonwealth/blob/master/knowledge_base/Devnet.md#manually-testing-evmos-dev-ethermint-sandbox) -> expect Keplr "Sign Ethereum" TXs to succeed
  - go to evmos-dev/proposals -> expect proposal to load, no console errors
-  negative test:
  - repeat above steps with a different account
  - expect proposal TX to fail, but previously created proposal still loads (node is not marked as "failed" due to failed tx)

For 6584:
- `update "ChainNodes" set slip44 = null where cosmos_chain_id = 'planq';` in local DB
- go to /planq/proposals
- (may need to refresh page here for next step to work)
- sign in 
  - expect metamask and keplr as wallet options
  - expect Keplr "Sign Ethereum" to work

## Deployment Plan

1. 

## Other Considerations

- Discovered a separate bug, ticketed: https://github.com/hicommonwealth/commonwealth/issues/7084
  - This bug is potential fixed as a side-effect of this PR.

### CODE REVIEW

**Review by timolegros — CHANGES_REQUESTED** [review]
2 small changes otherwise LGTM!

File: packages/commonwealth/server/util/cosmosProxy.ts [hunk 1]
```diff
@@ -295,6 +311,44 @@ function setupCosmosProxy(
...
+        if (!chainNode) return;
+        chainNode.slip44 = registeredChain.data.chain.slip44;
+        log.trace(
+          `Registered chain found for ${cosmos_chain_id}. slip44 recorded: ${registeredChain?.data?.chain?.slip44}`,
+        );
+        await chainNode.save();
+      }
+    } catch (err) {
+      // don't need to throw here, just trying to update slip44 if available
+      log.trace(
```
[the remark this answers is not in the document]
Comment (mhagel) [reply packages/commonwealth/server/util/cosmosProxy.ts]: done

File: packages/commonwealth/client/scripts/controllers/app/webWallets/utils.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { BalanceType } from '@hicommonwealth/core';
+import app from 'state';
+
+export const getCosmosEvmChains = (): string[] =>
```
[the remark this answers is not in the document]
Comment (mhagel) [reply packages/commonwealth/client/scripts/controllers/app/webWallets/utils.ts]: good call. done

File: packages/commonwealth/server/util/cosmosProxy.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (timolegros) [reply packages/commonwealth/server/util/cosmosProxy.ts]: @mhagel Can you update all the other logs as well? We shouldn't `JSON.stringify` anything into the error message anymore. Pino (our new logger) uses `fast-json-stringify` which is much faster than JSON.stringify. Any data that is passed to JSON.stringify should instead be passed as a context object.

[the remark this answers is not in the document]
Comment (mhagel) [reply packages/commonwealth/server/util/cosmosProxy.ts]: done @timolegros

**Review by timolegros — DISMISSED** [review]

**Comment by timolegros** [comment]
@CowMuon looks like Mark may have already solved some of our issues with EVM Cosmos in this PR. Massive shame we didn't merge this. @ianrowan let's get some reviews on this. I will resolve the conflicts today.

**Review by Rotorsoft — APPROVED** [review]
LGTM

**Pull request merged**